integration/src/integration/java/pl/allegro/tech/hermes/integration/PublishingTest.java

```diff
@@ -319,4 +324,41 @@ public void shouldPublishAndConsumeMessageWithTraceId() {
         assertThat(response).hasStatus(Response.Status.CREATED);
         assertThat(remoteService.waitAndGetLastRequest()).hasHeaderValue("Trace-Id", traceId);
     }
+
+    @Test
+    public void shouldPublishAndConsumeMessageWithTraceAndSpanHeaders() {
```

Comment:
This test is very similar to ```JmsConsumingTest.shouldPublishAndConsumeJmsMessageWithTraceHeaders```. If possible it would be nice to make both test cases more concise to make duplication less obvious. If not then maybe we should share some code between these tests...